com.playeveryware.eos/Runtime/Core/EOS_SDK_Additions/FileRequestTransferWrapper.cs

```diff
@@ -0,0 +1,142 @@
+        }
+
+        #region Equality operator overloads
+
+        /*
+         * The following section contains methods that are used to seemlessly
+         * provide wrapper functionality for the classes that this class wraps,
+         * allowing the wrapping to be transparent when assigning values to
+         * instances of the wrapper, or when comparing the instance to null (for
+         * instance in scenarios where the instance being wrapped is null, but
+         * the wrapper is not null - in those cases we still want to say it's
+         * equal to null to be as transparent of a wrapper as possible).
+         */
+
+        public static bool operator ==(FileRequestTransferWrapper<T> wrapper, object obj)
+        {
+            if (wrapper is null || wrapper._instance is null)
+            {
+                return obj is null;
+            }
+
+            return wrapper._instance.Equals(obj);
+        }
+
+        public static bool operator !=(FileRequestTransferWrapper<T> wrapper, object obj)
+        {
+            return !(wrapper == obj);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(this, obj)) return true;
+            if (ReferenceEquals(_instance, obj)) return true;
+
+            if (obj is FileRequestTransferWrapper<T> otherWrapper)
+            {
+                return Equals(_instance, otherWrapper._instance);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return _instance != null ? _instance.GetHashCode() : 0;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Used to call the CancelRequest function on the wrapped instance.
+        /// </summary>
+        /// <returns>
+        /// The result of canceling the wrapped request.
+        /// </returns>
+        public abstract Result CancelRequest();
+
+        /// <summary>
+        /// In a 1-to-1 wrapping, this method would simply call the "Release"
+        /// method on the wrapped class. However, the Dispose method is used
+        /// instead (and comments are included in the abstract DisposeInstance()
+        /// method that instruct implementing classes to call the Release()
+        /// method on the wrapped instance.
+        /// </summary>
+        public void Release()
+        {
+            Dispose();
+        }
+
+        /// <summary>
+        /// Used to neatly dispose of the wrapper and the instance that it
+        /// wraps.
+        /// </summary>
+        public void Dispose()
+        {
+            if (_instance == null)
+            {
+                return;
+            }
+
+            DisposeInstance();
+            _instance = null;
+        }
+
+        /// <summary>
+        /// When implementing this function, make sure to call the "Release"
+        /// method on the wrapped instance.
+        /// </summary>
+        protected abstract void DisposeInstance();
```

Comment:
Could this be virtual with a default implementation of releasing the instance? Should this be "DisposeInstance" or "ReleaseInstance"?

Reply:
Unfortunately, due to the nature of the FileRequestTransferWrapper being a generic function, and the type of the `_instance` variable not being known at compile-time, the implementing class must call it, because otherwise there is no way to know that `_instance.release()` is a function that can be called.